packages/aws-cdk/test/integ/bootstrap/bootstrap.integ-test.ts

```diff
@@ -0,0 +1,150 @@
+import * as core from '@aws-cdk/core';
+import * as cxapi from '@aws-cdk/cx-api';
+import * as AWS from 'aws-sdk';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { bootstrapEnvironment, deployStack, destroyStack, ISDK, loadToolkitInfo, SDK } from '../../../lib/api';
+import { bootstrapEnvironment2 } from '../../../lib/api/bootstrap/bootstrap-environment2';
+import { ExampleAsset, MyTestCdkStack } from './example-cdk-app/my-test-cdk-stack';
+
+jest.setTimeout(600_000);
+
+describe('Bootstrapping', () => {
+  const bootstrapStackName = 'AwsCdkBootstrapIntegTestLegacy';
+
+  const account = requireEnvVariable('TEST_ACCOUNT');
+  const region = requireEnvVariable('TEST_REGION');
+  let s3: AWS.S3;
+  let env: cxapi.Environment;
+  let sdk: ISDK;
+
+  beforeAll(() => {
+    s3 = new AWS.S3();
+    env = {
+      name: 'aws-cdk-bootstrap-integ-test',
+      account,
+      region,
+    };
+    sdk = new SDK({
+      userAgent: 'aws-cdk-bootstrap-integ-test',
+    });
+  });
+
+  describe('deploys the legacy bootstrap stack', () => {
+    const legacyBootstrapBucketName = 'aws-cdk-bootstrap-integ-test-legacy-bckt';
+    const newBootstrapBucketName = 'aws-cdk-bootstrap-integ-test-v2-bckt';
+    const exampleAppStack = 'BootstrapIntegTestExampleCdkAppStack';
+    const outdir = path.join(__dirname, 'cdk.out');
+
+    let bootstrapStack: cxapi.CloudFormationStackArtifact;
+    let testStack: cxapi.CloudFormationStackArtifact;
+
+    beforeAll(async () => {
+      // bootstrap the "old" way
+      const bootstrapResults = await bootstrapEnvironment(env, sdk, bootstrapStackName, undefined, {
+        bucketName: legacyBootstrapBucketName,
+      });
+      bootstrapStack = bootstrapResults.stackArtifact;
+    });
+
+    test('and then can deploy a CDK app using that bootstrap stack', async () => {
+      testStack = await deployCdkApp(outdir, env, sdk, bootstrapStackName, (app) => {
+        new MyTestCdkStack(app, exampleAppStack, {
+          assetType: ExampleAsset.ASSET_1,
+        });
+      });
+    });
+
+    describe('and then updates the bootstrap stack with the new resources', () => {
+      beforeAll(async () => {
+        // bootstrap the "new" way
+        const bootstrapResults = await bootstrapEnvironment2(env, sdk, bootstrapStackName, undefined, {
+          bucketName: newBootstrapBucketName,
+          trustedAccounts: ['790124522186', '593667001225'],
+          cloudFormationExecutionPolicies: [
+            'arn:aws:iam::aws:policy/AdministratorAccess',
+            'arn:aws:iam::aws:policy/AmazonS3FullAccess',
```

Comment:
Is this not subsumed by AdministratorAccess?

Reply:
Yeah, it is. Those roles are not used in the test in any way, I just wanted to test the case with more than one argument for the parameter. So I grabbed the first 2 managed policies that I could think of.

Reply:
ack